### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by oschwald**
This pull request contains several performance improvements. Most of the gains
come from the first commit.

While working on this, I also noticed a bug in decodeInt when size is greater
than 0 and less than 4. I fixed this and added some test. If you are open to it,
I'd be happy to add all of the tests seen in the official readers for the type
decoding. The original decodeUint also had some issues with sizes between 4 and 8,
for instance.

I also wasn't sure why the reader converts big ints to strings. I matched this in
my changes, but it seems like it would be better to return them directly.

### CODE REVIEW

**Comment by runk** [comment]
It looks great! Thanks for doing it 🙇 

I noticed there's some minor formatting differences - could you please run `npm run format`?

I've raised https://github.com/runk/mmdb-lib/pull/226 to have formatting check as part of CI pipeline, it'd be great if you'd check it

File: src/decoder.ts [hunk 1]
```diff
@@ -280,40 +266,31 @@ export default class Decoder {
...
+    }
+    if (size > 16) {
+      return 0
+    }
+    return this.decodeBigUint(offset, size);
   }
 
+
   private decodeString(offset: number, size: number) {
-    return this.db.slice(offset, offset + size).toString();
```
Comment (runk) [inline src/decoder.ts:316]: Nice. Not sure why I did it that way in first place.

**Comment by oschwald** [comment]
Woops, I missed the format script entirely. I'd be happy to run that on each commit. It probably makes sense for your PR to merged first as there are a couple of existing issues in `master`.

**Review by runk — APPROVED** [review]

**Comment by runk** [comment]
@oschwald FYI to trigger release, commit messages need to be formatted according to semantic release rules: https://github.com/semantic-release/semantic-release?tab=readme-ov-file#commit-message-format

I had to do https://github.com/runk/mmdb-lib/actions/runs/14743180230 to release your change

Not a big deal, but always nice when it's going out automatically

**Comment by oschwald** [comment]
Thanks. I'll try to remember that when committing to your repos in the future.

What do you think about adding a full suite of decoder tests similar to [these in the PHP reader](https://github.com/maxmind/MaxMind-DB-Reader-php/blob/main/tests/MaxMind/Db/Test/Reader/DecoderTest.php)? I suspect there may be a couple of other bugs lingering for some of the edge cases, e.g., long strings. There aren't test databases for all these cases as some of them would end up being rather large.

**Comment by runk** [comment]
More tests like those are always great 👍
